django/db/models/query.py

```diff
@@ -1566,12 +1566,13 @@ def prefetch_one_level(instances, prefetcher, lookup, level):
     #  callable that gets value to be matched for returned instances,
     #  callable that gets value to be matched for passed in instances,
     #  boolean that is True for singly related objects,
-    #  cache name to assign to).
+    #  cache or field name to assign to,
+    #  boolean that is True when the previous argument is a cache name vs a field name).
 
     # The 'values to be matched' must be hashable as they will be used
     # in a dictionary.
 
-    rel_qs, rel_obj_attr, instance_attr, single, cache_name = (
+    rel_qs, rel_obj_attr, instance_attr, single, cache_name, is_cache = (
```

Comment:
Sadly yes https://github.com/django/django/pull/6721/files#diff-50229e6c26e18f27542f698383cdaccfR212.

Without this it's not possible to know when we are passed a field name (that points to descriptor) or the cache name.